prow/plugins/welcome/welcome.go

```diff
@@ -112,20 +112,29 @@ func handlePR(c client, t plugins.Trigger, pre github.PullRequestEvent, welcomeT
 		return nil
 	}
 
+	org := pre.PullRequest.Base.Repo.Owner.Login
+	repo := pre.PullRequest.Base.Repo.Name
+	user := pre.PullRequest.User.Login
+	pullRequestNumber := pre.PullRequest.Number
+
+	log := c.Logger.WithFields(logrus.Fields{"org": org, "repo": repo, "number": pullRequestNumber})
+
 	// ignore bots, we can't query their PRs
 	if pre.PullRequest.User.Type != github.UserTypeUser {
+		log.Infof("Ignoring bot user '%s', as querying their PRs is not possible", user)
 		return nil
 	}
 
-	org := pre.PullRequest.Base.Repo.Owner.Login
-	repo := pre.PullRequest.Base.Repo.Name
-	user := pre.PullRequest.User.Login
+	if alwaysPost {
+		log.Infof("Welcome plugin is configured to always post messages")
```

Comment:
Instead of this log I think an else branch of the `if alwaysPost || len(issues) == 0 || len(issues) == 1 && issues[0].Number == pre.Number` statement saying something like "ignoring the pr because xyz" would be appropriate

Reply:
Sure thing, added a log line in the relevant else branch